lib/bolt/config/options.rb

```diff
@@ -509,6 +509,10 @@ module Options
         "remote" => {
           description: "A map of configuration options for the remote transport.",
           type: Hash,
+          additionalProperties: {
+            description: "Parameter to pass to the remote task",
+            _plugin: true
+          },
```

Comment:
```suggestion
          additionalProperties: true,
```

JSON Schema Draft 07 lets `additionalProperties` be a boolean. By default it's set to `true`, but we can explicitly set it to `true` to tell the validator to skip the `validate_keys` check.

Then the validator just needs a small update [here](https://github.com/puppetlabs/bolt/blob/main/lib/bolt/validator.rb#L90):

```
elsif definition[:additionalProperties].is_a?(Hash)
```

That'll let us skip the extra logic in the spec test, too.